Content.Server/Storage/Components/StorageFillComponent.cs

```diff
@@ -51,10 +52,21 @@ void IMapInit.MapInit()
 
                 for (var i = 0; i < storageItem.Amount; i++)
                 {
-                    storage.Insert(Owner.EntityManager.SpawnEntity(storageItem.PrototypeId, Owner.Transform.Coordinates));
+                    storage.Insert(
+                        Owner.EntityManager.SpawnEntity(storageItem.PrototypeId, Owner.Transform.Coordinates));
                 }
+
                 if (!string.IsNullOrEmpty(storageItem.GroupId)) alreadySpawnedGroups.Add(storageItem.GroupId);
             }
+
+            // Finished Inserting all
+            // disregard poor quality of code
+            if (Owner.TryGetComponent(out ServerStorageComponent? container))
+            {
+                var contained = container.StoredEntities ?? new List<IEntity>();
+                Owner.EntityManager.EventBus.RaiseEvent(EventSource.Network,
+                    new AfterStorageFillEvent(container.Owner,  contained));
+            }
```

Comment:
We really don't need a network event for this, and if we did you pass EntityUids over the network not IEntity.